### PR TITLE
[Leptonica] Bump to 1.83.1

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: L/Leptonica/build_tarballs.jl [hunk 1]
```diff
@@ -46,12 +47,12 @@ dependencies = [
     Dependency(PackageSpec(name="JpegTurbo_jll", uuid="aacddb02-875f-59d6-b918-886e6ef4fbf8")),
     Dependency(PackageSpec(name="libpng_jll", uuid="b53b4c65-9356-5827-b1ea-8c7a1a84506f")),
     Dependency("Libtiff_jll"; compat="4.3.0"),
```
Comment (giordano) [inline L/Leptonica/build_tarballs.jl:49]: Use newer libtiff?

File: L/Leptonica/build_tarballs.jl [hunk 2]
```diff
@@ -46,12 +47,12 @@ dependencies = [
...
-    Dependency(PackageSpec(name="libwebp_jll", uuid="c5f90fcd-3b7e-5836-afba-fc50a0988cb2")),
+    Dependency("libwebp_jll"; compat="^1.2.4"),
```
Comment (giordano) [inline L/Leptonica/build_tarballs.jl:50]: `^` is the default anyway 
```suggestion
    Dependency("libwebp_jll"; compat="1.2.4"),
```

File: L/Leptonica/build_tarballs.jl [hunk 3]
```diff
@@ -46,12 +47,12 @@ dependencies = [
...
     Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a")),
     # Leptonica has a runtime check on the minor version of OpenJpeg, because why not:
     # https://github.com/DanBloomberg/leptonica/blob/68d2cc15b955192f65772689d258a6d10dba52f5/src/jp2kio.c#L268-L272
     Dependency("OpenJpeg_jll"; compat="~2.4.0"),
```
Comment (giordano) [inline L/Leptonica/build_tarballs.jl:54]: Use newer openjpeg?

File: L/Leptonica/build_tarballs.jl [hunk 4]
```diff
@@ -27,7 +28,7 @@ platforms = supported_platforms()
 
 # The products that we will ensure are always built
 products = [
-    LibraryProduct("liblept", :liblept),
+    LibraryProduct("libleptonica", :liblept),
```
Comment (giordano) [inline L/Leptonica/build_tarballs.jl:31]: The change of the soname will actively break dependents.  It'd be good to fix compat bounds of dependents jll before moving forward with this PR.

**Review by giordano — CHANGES_REQUESTED** [review]

File: L/Leptonica/build_tarballs.jl (hunk 1 above)
Comment (jeremiahpslewis) [inline L/Leptonica/build_tarballs.jl:49]:
```suggestion
    Dependency("Libtiff_jll"; compat="4.5.1"),
```

File: L/Leptonica/build_tarballs.jl [hunk 5]
```diff
@@ -45,13 +46,13 @@ dependencies = [
     Dependency(PackageSpec(name="Giflib_jll", uuid="59f7168a-df46-5410-90c8-f2779963d0ec")),
     Dependency(PackageSpec(name="JpegTurbo_jll", uuid="aacddb02-875f-59d6-b918-886e6ef4fbf8")),
     Dependency(PackageSpec(name="libpng_jll", uuid="b53b4c65-9356-5827-b1ea-8c7a1a84506f")),
-    Dependency("Libtiff_jll"; compat="4.3.0"),
-    Dependency(PackageSpec(name="libwebp_jll", uuid="c5f90fcd-3b7e-5836-afba-fc50a0988cb2")),
+    Dependency("Libtiff_jll"; compat="4.5.1"),
```
Comment (jeremiahpslewis) [inline L/Leptonica/build_tarballs.jl:49]:
```suggestion
    Dependency("Libtiff_jll"; compat="~4.5.1"),
```

**Review by giordano — APPROVED** [review]

**Pull request merged**